zeebe/backup/src/main/java/io/camunda/zeebe/backup/metrics/CheckpointMetrics.java

```diff
@@ -7,57 +7,56 @@
  */
 package io.camunda.zeebe.backup.metrics;
 
-import io.prometheus.client.Counter;
-import io.prometheus.client.Gauge;
+import static io.camunda.zeebe.backup.metrics.CheckpointMetricsDoc.*;
 
-public class CheckpointMetrics {
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.EnumMap;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
 
-  private static final String NAMESPACE = "zeebe";
-  private static final String LABEL_NAME_PARTITION = "partition";
-  private static final String LABEL_NAME_RESULT = "result";
-
-  private static final Counter CHECKPOINT_RECORDS =
-      Counter.build()
-          .namespace(NAMESPACE)
-          .name("checkpoint_records_total")
-          .help(
-              "Number of checkpoint records processed by stream processor. Processing can result in either creating a new checkpoint or ignoring the record. This can be observed by filtering for label 'result'.")
-          .labelNames(LABEL_NAME_RESULT, LABEL_NAME_PARTITION)
-          .register();
-
-  private static final Gauge CHECKPOINT_POSITION =
-      Gauge.build()
-          .namespace(NAMESPACE)
-          .name("checkpoint_position")
-          .help("Position of the last checkpoint")
-          .labelNames(LABEL_NAME_PARTITION)
-          .register();
-
-  private static final Gauge CHECKPOINT_ID =
-      Gauge.build()
-          .namespace(NAMESPACE)
-          .name("checkpoint_id")
-          .help("Id of the last checkpoint")
-          .labelNames(LABEL_NAME_PARTITION)
-          .register();
-
-  final String partitionId;
-
-  public CheckpointMetrics(final int partitionId) {
-    this.partitionId = String.valueOf(partitionId);
+public class CheckpointMetrics {
+  private final MeterRegistry registry;
+  private final EnumMap<CheckpointMetricsResultValue, Counter> checkpointRecords =
```

Comment:
:wrench: Same as before, let's use the least specific type for field declaration.